hems/shm/shm.go

```diff
@@ -50,63 +49,53 @@ type SEMP struct {
 	site         site.API
 }
 
-// New generates SEMP Gateway listening at /semp endpoint
-func New(conf map[string]interface{}, site site.API, httpd *server.HTTPd) (*SEMP, error) {
-	cc := struct {
-		VendorID     string
-		DeviceID     string
-		AllowControl bool
-	}{
-		VendorID: "28081973",
-	}
-
-	if err := util.DecodeOther(conf, &cc); err != nil {
-		return nil, err
+func New(allowControl bool, vendorId, deviceId string, site site.API, addr string, router *mux.Router) (*SEMP, error) {
+	if vendorId == "" {
```

Comment:
Can we keep the usual `NewFromConfig` pattern? Move the config structure to this package.